modules/public/logic.py

```diff
@@ -0,0 +1,258 @@
+            percentage_of_users = country_info['percentage_of_users'] or 0
+            color_rgb = country_info['color_rgb'] or [247, 247, 247]
+
+        # Use common_name if available to be less political offending (#310)
+        try:
+            country_name = country.common_name
+        except AttributeError:
+            country_name = country.name
+
+        country_data[country.numeric] = {
+            'name': country_name,
+            'code': country.alpha_2,
+            'percentage_of_users': percentage_of_users,
+            'color_rgb': color_rgb
+        }
+
+        if display_number_users:
+            country_data[country.numeric]['number_of_users'] = number_of_users
+
+    return country_data
+
+
+def get_searched_snaps(search_results):
+    """Get search snaps from API response
+
+    :param search_results: the body responsed by the API
+
+    :returns: The list of the searched snaps
+    """
+    return (
+        search_results['_embedded']['clickindex:package']
+        if '_embedded' in search_results
+        else []
+    )
+
+
+def get_pages_details(url, links):
+    """Transform returned navigation links from search API from limit/offset
+    to size/page
+
+    :param url: The url to build
+    :param links: The links returned by the API
+
+    :returns: A dictionnary with all the navigation links
+    """
+    links_result = {}
+
+    if('first' in links):
+        links_result['first'] = convert_limit_offset_to_size_page(
+            url,
+            links['first']['href']
+        )
+
+    if('last' in links):
+        links_result['last'] = convert_limit_offset_to_size_page(
+            url,
+            links['last']['href']
+        )
+
+    if('next' in links):
+        links_result['next'] = convert_limit_offset_to_size_page(
+            url,
+            links['next']['href']
+        )
+
+    if('prev' in links):
+        links_result['prev'] = convert_limit_offset_to_size_page(
+            url,
+            links['prev']['href']
+        )
+
+    if('self' in links):
+        links_result['self'] = convert_limit_offset_to_size_page(
+            url,
+            links['self']['href']
+        )
+
+    return links_result
+
+
+def convert_limit_offset_to_size_page(url, link):
+    """Convert navigation link from offst/limit to size/page
+
+    :param url: The new url
+    :param link: The navigation url returned by the API
+
+    :returns: The new navigation link
+    """
+    url_parsed = urlparse(link)
+    host_url = (
+        "{base_url}"
+        "?q={q}&limit={limit}&offset={offset}"
+    )
+
+    url_queries = parse_qs(url_parsed.query)
+    q = url_queries['q'][0]
+    size = int(url_queries['size'][0])
+    page = int(url_queries['page'][0])
+
+    return host_url.format(
+        base_url=url,
+        q=q,
+        limit=size,
+        offset=size*(page-1)
+    )
+
+
+def format_paragraphs(unformatted_description):
+    """Format paragraphs to displayable paragraph
```

Comment:
"Split a block of text into paragraphs, assuming 2 or more line breaks signifies a paragraph break."